exonum-java-binding/qa-service/src/test/java/com/exonum/binding/qaservice/QaServiceImplIntegrationTest.java

```diff
@@ -114,356 +88,394 @@ void tearDown() {
+    }
   }
 
   @Test
-  @RequiresNativeLibrary
   void getTxMessages() {
-    withNodeFake(() -> {
+    try (TestKit testKit = TestKit.forService(QaServiceModule.class)) {
+      QaServiceImpl service = testKit.getService(QaService.ID, QaServiceImpl.class);
+      TransactionMessage createCounterTransaction = createCreateCounterTransaction("counterName");
+      testKit.createBlockWithTransactions(createCounterTransaction);
       Map<HashCode, TransactionMessage> txMessages = service.getTxMessages();
-      assertThat(txMessages).isEmpty();
-    });
+      assertThat(txMessages).hasSize(2);
+      assertThat(txMessages.get(createCounterTransaction.hash()))
+          .isEqualTo(createCounterTransaction);
+    }
   }
 
   @Test
-  @RequiresNativeLibrary
   void getTxResults() {
-    withNodeFake(() -> {
+    try (TestKit testKit = TestKit.forService(QaServiceModule.class)) {
+      QaServiceImpl service = testKit.getService(QaService.ID, QaServiceImpl.class);
+      TransactionMessage createCounterTransaction = createCreateCounterTransaction("counterName");
+      testKit.createBlockWithTransactions(createCounterTransaction);
       Map<HashCode, TransactionResult> txResults = service.getTxResults();
-      assertThat(txResults).isEmpty();
-    });
+      assertThat(txResults).hasSize(1);
+      assertThat(txResults.get(createCounterTransaction.hash()))
+          .isEqualTo(TransactionResult.successful());
+    }
   }
 
   @Test
-  @RequiresNativeLibrary
   void getTxResult() {
-    withNodeFake(() -> {
-      HashCode messageHash = HashCode.fromBytes(createPrefixed(bytes(0x00), 32));
-      Optional<TransactionResult> txResult = service.getTxResult(messageHash);
-      assertThat(txResult).isEmpty();
-    });
+    try (TestKit testKit = TestKit.forService(QaServiceModule.class)) {
+      QaServiceImpl service = testKit.getService(QaService.ID, QaServiceImpl.class);
+      TransactionMessage createCounterTransaction = createCreateCounterTransaction("counterName");
+      testKit.createBlockWithTransactions(createCounterTransaction);
+      Optional<TransactionResult> txResult = service.getTxResult(createCounterTransaction.hash());
+      assertThat(txResult).contains(TransactionResult.successful());
+    }
   }
 
   @Test
-  @RequiresNativeLibrary
   void getTxLocations() {
-    withNodeFake(() -> {
+    try (TestKit testKit = TestKit.forService(QaServiceModule.class)) {
+      QaServiceImpl service = testKit.getService(QaService.ID, QaServiceImpl.class);
+      TransactionMessage createCounterTransaction = createCreateCounterTransaction("counterName");
+      testKit.createBlockWithTransactions(createCounterTransaction);
       Map<HashCode, TransactionLocation> txLocations = service.getTxLocations();
-      assertThat(txLocations).isEmpty();
-    });
+      assertThat(txLocations).hasSize(1);
+      TransactionLocation expectedTransactionLocation = TransactionLocation.valueOf(1L, 0L);
```

Comment:
Is it correct that TransactionLocation has index in the block as long (we can't usually have more than a couple of thousands transactions per block, but > 2^31-1 is impossible).

Why do we have `uint64 position_in_block = 2;`?

Reply:
In core we have `position_in_block: u64`

Reply:
> Is it correct that TransactionLocation has index in the block as long (we can't usually have more than a couple of thousands transactions per block, but > 2^31-1 is impossible).

@aleksuss Is there a reason for this? Why don't we use `uint32` — it will make client code clearer?